server/backend/api/app/routers/router_application.py

```diff
@@ -43,3 +49,36 @@ def get_fam_application_user_role_assignment(
                  # of results = {len(app_user_role_assignment)}"
     )
     return app_user_role_assignment
+
+
+@router.get(
+    "/{application_id}/users/{user_id}",
+    response_model=FamUserInfoSchema,
+    status_code=200,
+    dependencies=[Depends(authorize_by_app_id)],
+    summary="Retrieve User Information by User ID under an application",
+    responses={
+        404: {
+            "description": "User not found",
+            "content": {"application/json": {"example": {"detail": "User not found"}}},
+        }
+    },
+)
+async def get_user_by_user_id(
+    user_id: int,
+    db: Session = Depends(database.get_db),
+):
+    """
+    Retrieve the user data for a given user id under an authorized application.
+
+    Args:
+        userId (int): The ID of the user.
+        applicationId (int): The ID of the application the user has access to.
+
+    Returns:
+        FamUserInfoSchema: The user information corresponding to the provided userId.
+    """
+    user = crud_user.get_user(user_id=user_id, db=db)
```

Comment:
For now, this endpoint might be ok.
But the "applicationId" isn't being used in finding the user within the application which is odd (only being used in validation of the requester), so it does not feel like a proper place for this endpoint to me.

Can we later improve this logic by either adding a filtering on "applicationId" (crud) or migrate this endpoint to "server/admin_management" with router_user (and with proper validation logic)?

Reply:
You’re right. My priority was to get an endpoint up and running quickly by leveraging existing CRUD functions, as this wasn’t initially planned. In the future, we should definitely add logic to verify if the user actually belongs to the specified application to ensure proper validation.

Reply:
created a ticket here
https://github.com/bcgov/nr-forests-access-management/issues/1613

Reply:
Thanks!